src/backend/distributed/planner/multi_explain.c

```diff
@@ -1453,6 +1473,17 @@ WrapQueryForExplainAnalyze(const char *queryString, TupleDesc tupleDesc)
 	 * number of columns returned by worker_save_query_explain_analyze.
 	 */
 	char *workerSaveQueryFetchCols = (tupleDesc->natts == 0) ? "" : "*";
+
+	if (params != NULL)
+	{
+		/*
+		 * Add a dummy CTE to ensure all parameters are referenced, such that their
+		 * types can be resolved.
+		 */
+		appendStringInfo(wrappedQuery, "WITH unused AS (%s) ",
```

Comment:
Just wondering, if we should push this part of the query too into the routine and have one routine (GenerateParamResDummyCTE()) that builds the dummy CTE from passed-params